src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -286,7 +287,43 @@ app.directive('dashboardApp', function ($injector) {
         }).catch(notify.fatal);
       };
 
+      $scope.showFilterBar = () => filterBar.getFilters().length > 0 || !$scope.fullScreenMode;
+      let onRouteChange;
+      const setFullScreenMode = (fullScreenMode) => {
+        $scope.fullScreenMode = fullScreenMode;
+        dashboardState.setFullScreenMode(fullScreenMode);
+        chrome.setVisible(!fullScreenMode);
+        $scope.$root.$broadcast('globalNav:update');
+        if (fullScreenMode) {
+          onRouteChange = $scope.$on('$routeChangeStart', () => chrome.setVisible(true));
```

Comment:
I'll add a comment to make it clearer.  I need to add this so that if the user navigates to a different app, or clicks the back button, the user is bounced out of full screen mode.  Otherwise, you can end up with something like this:
<img width="1769" alt="screen shot 2017-06-21 at 2 33 57 pm" src="https://user-images.githubusercontent.com/16563603/27400616-aff0d2dc-568e-11e7-941a-c696dd081c8b.png">